pkg/recipe/variable.go

```diff
@@ -167,19 +180,48 @@ func (val VariableValues) Validate() error {
 	return nil
 }
 
+func (r *VariableValidator) CreateTableValidatorFunc() func(cols []string, rows [][]string, input string) error {
+	if r.Unique {
+		return func(cols []string, rows [][]string, input string) error {
+			colIndex := slices.Index(cols, r.Column)
+			colValues := make([]string, len(rows))
+			for i, row := range rows {
+				colValues[i] = row[colIndex]
+			}
+			slices.Sort(colValues)
+
+			if uniqValues := len(slices.Compact(colValues)); uniqValues != len(colValues) {
+				if r.Help != "" {
+					return errors.New(r.Help)
+				} else {
+					return errors.New("value not unique within column")
+				}
+			}
+
+			return nil
+		}
+	}
+
+	panic(fmt.Errorf("unsupported table validator on column %q", r.Column))
```

Comment:
Added these but not sure if needed or is everything validated to the extent that this will never happen? But I also noticed that there is no validation that a validator is applied to a column that exists. Maybe there should be as now it looks like all is good but nothing is validated if there is eg. a typo in `column` property.

Reply:
Yeah, we could add these invalid validator errors here. But we could try to return an error instead of panic, so the signature would be something like this:  `CreateTableValidatorFunc() (func(cols []string, rows [][]string, input string) error, error)`. It's a quite mouthful, but at least we avoid `panic` :)

Reply:
Should the error still be just ignored or traverse up the call tree where it would go quite far up? Now the recipe validation fails with empty regex pattern if defining something like
```yaml
validators:
  - kissa: true
    column: KEIJO
```

Reply:
Validator column typos are checked here: https://github.com/futurice/jalapeno/blob/50235f687244f6158403f2f0f9638b168f0127dc/pkg/recipe/variable.go#L137-L153

Reply:
(That `kissa: true` is just ignored)

Reply:
> Yeah, we could add these invalid validator errors here. But we could try to return an error instead of panic, so the signature would be something like this: `CreateTableValidatorFunc() (func(cols []string, rows [][]string, input string) error, error)`. It's a quite mouthful, but at least we avoid `panic` :)

I now added these but don't really handle the errors apart from just silently skipping the validator. Technically this should happen only if a programming error is made as the validators are somehow validated in the variable level validation.

Other way would have been like it was with the regex earlier that if the pattern compiles a function is returned and that's it. So if the non-table validator would always return regex validator and the table one would always return the unique validator things would kinda work but IMO that would not be very good.

Reply:
> (That `kissa: true` is just ignored)

Oh and this is not the case, there is an error from regex pattern:

```yaml
  - name: STRING_VAR_2
    description: |
      Simple string variable
    validators:
      - kissa: true
```

leads to

```
➜  jalapeno git:(feat/unique-column) ✗ task run -- execute ./examples/variable-types
task: [run] go run /Users/vesa.poikajarvi/w/futurice/jalapeno/cmd/jalapeno execute ./examples/variable-types

Error: can not load the recipe: loaded recipe was invalid: error on variable STRING_VAR_2: validator 1: regexp pattern is empty
exit status 1
```

Reply:
I mean that these behave similarly:
```
- name: STRING_VAR_2
  description: |
    Simple string variable
  validators:
    - kissa: true
```
and
```
  - name: STRING_VAR_2
    description: |
      Simple string variable
    validators:
      - {}
```

Anyway the error is correct, that the validator is invalid